modules/apps/asset/asset-publisher-web/src/main/resources/META-INF/resources/view.jsp

```diff
@@ -33,10 +33,14 @@ if (assetPublisherDisplayContext.isEnableTagBasedNavigation() && !assetPublisher
 
 			<div class="btn-group-item">
 				<clay:link
-					elementClasses="btn btn-outline-borderless btn-outline-secondary btn-sm"
+					borderless="<%= true %>"
+					button="<%= true %>"
```

Comment:
The API changes here are more obvious. We replace `elementClasses` with named attributes shared with `clay:button`.

The only problem is we need to have a way to distinguish between `link-` and `btn-` for the css class prefixes, thus `button="<%= true %>"`.

Another option would be... `notSureAboutTheName="link|button"`. Couldn't come up with a good name for this though, thus `button="<%= true %>"`

Reply:
`type="link|button"` maybe?

Reply:
TIL, anchors do have a native `type` attribute, so we probably don't want to overload that: https://html.spec.whatwg.org/#dom-a-type

Reply:
Either of those sound fine to me. I think `type=` sounds more future-proof than a boolean. `ClayButton` has a `type` property too (default: 'button').

Reply:
![tenor-90920564](https://user-images.githubusercontent.com/156388/88547907-bd1d3800-cff4-11ea-8160-43f1817bf4ad.gif)

Ah right, and people can pass it via the dynamic attributes mechanism right

Reply:
Yeah, so my concern is, would `<a href="foo" type="button">Hey you</a>` be correct [according to the spec](https://html.spec.whatwg.org/#attr-hyperlink-type)?

Reply:
> The value must be a valid MIME type string

Reply:
Sorry, I'm rambling

> Ah right, and people can pass it via the dynamic attributes mechanism right

This pointed by @brunobasto is the key. If we overload the `type` attribute, then it wouldn't be possible to pass it any other way to the underlying html

Reply:
But you wouldn't just blindly pass through and print `type`, would you?

Reply:
So, if we want to give it a different meaning, then it becomes a tag attribute and part of the API. Right now, we don't have to define it and people can still pass values down to the anchor element.

So, right now we can do:

```
<clay:link href="foo" type="text/html" />
<a href="foo" type="text/html" />
```

If we overload `type`, then we would add it to the `clay.tld` definition, and then we'll have to define the rules for what is our API and what isn't, so:

```
<clay:link href="foo" type="button" />
<a class="btn" href="foo" />

<clay:link href="foo" type="link" />
<a href="foo" />
```

And

```
<clay:link href="foo" type="text/html" />
<a href="foo type="text/html" />

<!-- custom syntax? -->
<clay:link href="foo" type="button:text/html" />
<a class="btn" href="foo" type="text/html" />
```

Reply:
> The type attribute, if present, gives the MIME type of the linked resource. It is purely advisory. The value must be a valid MIME type string. User agents must not consider the type attribute authoritative — upon fetching the resource, user agents must not use metadata included in the link to the resource to determine its type.

I think it should be fine to repurpose it though. Is the Clay component gonna render this type attribute or will it just discard it after the first render?

Reply:
> https://github.com/liferay/clay/blob/master/packages/clay-link/src/index.tsx#L97 I think it will pass it through right

We're not even using `ClayLink`. I see no immediate reason to hydrate an anchor tag, so we're only dealing with the Java output here.

Reply:
Ah. Got it. I assumed we were using it. Anyway I think we have a valid argument for not overloading the `type` attribute and going with the boolean approach. Even if that type will most likely never be used the way the spec intended it to 🤣

Reply:
> that type will most likely never be used the way the spec intended it to

For me that observation is the key. AFAIK I know, pretty much nobody uses the `type` attribute, and [even if they do](https://stackoverflow.com/a/10508040/2103996):

>  it's purely advisory and user agents **must not** consider it authoritative. In other words, it's not even a guarantee that it's correct and is just a _hint_. I'd think that's based purely on practicality, since links can become stale very quickly; the link cannot dictate the content of the linked-to resource.

I think that avoiding impedance mismatches between the React API and the taglib API is desirable. @jbalsas pointed out in Slack, we could actually change the React side to match the taglib side (but that would require a deprecation cycle in order to do it in a backwards-compatible way). But overall I think the combination of:

- Rarity of usage of HTML `type` attribute.
- Desirability of parity across React and taglib implementations.
- Ability to implement a `typeAttribute` (or whatever we want to call it) property as an escape hatch to allow setting of `type` in the DOM, in case we find we ever need it (we never will though); we should make the common case easy and the rare case possible, but we shouldn't structure our API around the rare case.

Add up to `type=button|link` being the best trade-off. I get the sense that nobody feels really strongly about the options, so if you need a tie-breaker, consider that my vote.

Reply:
![tenor-155063225](https://user-images.githubusercontent.com/156388/88819840-e970ba00-d196-11ea-9b46-bb2dfb4b25f2.gif)